### DISCUSSION TITLE
[FEATURE] Stancer integration

### DISCUSSION THREAD

**Discussion by beaujean** [juspay/hyperswitch#2826] [Ideas / Feature Requests]
### Feature Description

Stancer is a new PSP and it desserve its connector in hyperswitch as it is a simple, reliable and cost efficient and transparent payment API. Stancer provides card and SEPA Direct Debit payments methods.
See https://www.stancer.com/

### Possible Implementation

Stancer has a clear API with great documentation available here https://wearestancer.gitlab.io/doc/en/API.html

I've started the implementation which is visible here https://github.com/beaujean/hyperswitch/tree/stancer

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Yes, I am willing to submit a PR!

**Comment by manojradhakrishnan** [comment]
Hi @beaujean 

We will be happy to accept your contribution of Stancer integration. It very well aligns with our values.

We already support cards across 50+ processors and SEPA direct debit with few processors like Stripe, Gocardless etc., Adding Stancer will be a great enhancement for the merchant community. 

Please let us know what support you need from us to start contributing. Once you are good to go, we can create an issue and assign it to you.

@SMohata @SamraatBansal

**Reply by beaujean** [reply]
Hi @manojradhakrishnan,

We already started to work on the implementation here https://github.com/beaujean/hyperswitch/tree/stancer do we need to continue under this projet in order to merge in `main` later or car you open a `stancer` branch on your side where we will submit step by step our features ?

Thank you for allowing us working together on this great initiative.

**Reply by manojradhakrishnan** [reply]
Great! Forking the repository and giving a pull request is a recommended approach. 

Recommended commit message guidelines and things to take care while submitting PRs are mentioned here: https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md

Once you are ready with a PR, please let us know. WE will create a corresponding issue and assign it to you.

**Reply by manojradhakrishnan** [reply]
Hi @beaujean - A similar issue opened for contributing a new connector integration. I thought it might be helpful for you.

https://github.com/juspay/hyperswitch/discussions/2944#discussioncomment-7713783

**Comment by NarsGNA** [comment]
Hey @beaujean, we are reviewing your PR and will revert over the week. Thank you for your patience! 
Tracking it here: [#2876 feat(connector)](https://github.com/juspay/hyperswitch/pull/2876)

**Comment by SamraatBansal** [comment]
Hey @beaujean, we are in the process of reviewing your PR and noticed your branch is behind and has conflicts with the `main` branch.
We have made some changes to the structure of the connector code as well, As and when you compile after merging with main you will be able to get and resolve the errors.
Please update your PR branch and we will merge accordingly.

Thanks for your efforts and patience.

**Comment by swangi-kumari** [comment]
Hello @beaujean ,
We want to test Stancer via Hyperswitch. However, we are facing trouble creating an account on your website. Could you please help us in creating an account so that we can obtain API keys to test it?"

**Reply by beaujean** [reply]
Hi, could you please write to support@stancer.com and we will help you in the subscription process. Thanks.

**Reply by swangi-kumari** [reply]
Sure, Thanks a lot!

**Comment by pierre-H** [comment]
Hi @beaujean ,
What is the status of the Stancer integration?

**Reply by pierre-H** [reply]
@beaujean any news ?

**Comment by gorakhnathy7** [comment]
Hey @pierre-H @beaujean 
The PR have been having a lot of conflicts lately, let me get it picked internally to be resolved and merged, 
Also, can you please help with the sandbox access to Stancer so we can test these thoroughly?

**Reply by pierre-H** [reply]
Hi @gorakhnathy7  @beaujean 
Any news ?
